### PR TITLE
Mail (#32)

### PR DESCRIPTION
**Pull request by Kavakuo**
Addresses #32 partly.

Sending a Mail when an account is verified manually depends on #34.

### CODE REVIEW

File: labshare-client/src/backend/endpoints/language.ts [hunk 1]
```diff
@@ -35,21 +38,18 @@ export function language(req: express.Request, res: express.Response) {
             }
         }
 
-        let selected_lang_id = selected_lang.language
-        if (selected_lang_id.indexOf('de') > -1) {
+        let tmp = selected_lang.language
+        if (tmp.indexOf('de') > -1) {
```
Comment (SethGreylyn) [inline labshare-client/src/backend/endpoints/language.ts:42]: `tmp` is completely unnecessary.

File: labshare-client/src/backend/endpoints/registration.ts [hunk 2]
```diff
@@ -36,21 +44,42 @@ export async function registration(req: express.Request, res: express.Response,
...
+    let user = new model(body)
+    
+    let token = uuid();
+    let token_doc = new ActivationToken({ token: token, objectId: user._id });
 
-    doc.save().then((doc) => {
-        if (!doc) {
-            return utils.internalError(res)
-        }
+    user.save().then((doc) => {
```
Comment (SethGreylyn) [inline labshare-client/src/backend/endpoints/registration.ts:71]: This promise block can be `awaited` and put in a `try/catch` without having to change anything major

File: labshare-client/src/backend/endpoints/resetPassword.ts [hunk 3]
```diff
@@ -18,6 +19,12 @@ export async function resetPassword(req: express.Request, res: express.Response,
     if (!token_doc) {
         return utils.badRequest(res);
     }
+
+    let oldest = new Date()
+    oldest.setDate(oldest.getDate() - 2)
+    if (token_doc.createdAt < oldest) {
```
Comment (SethGreylyn) [inline labshare-client/src/backend/endpoints/resetPassword.ts:25]: Is there a reason we're creating a `token_doc` and not using the token itself to know when it expires? Even if we want to persist the token, we should be able to just do something like `token_doc.token.exp < new Date()` to make sure we're not accepting expired tokens.

Comment (Kavakuo) [reply labshare-client/src/backend/endpoints/resetPassword.ts:25]: These tokens (activate, password reset) aren't JWTs. So I just generate a token using `uuid` and store it in the database. But you are right, we could JWTs for this as well, didn't think about it until now. Thanks!

**Pull request merged**